src/pages/deploy_application/configure_source.jsx

```diff
@@ -47,6 +47,7 @@ export default function ConfigureSourcePage({
   const uploadCodeFieldRef = useRef(null);
   const tarballRef = useRef(null);
   const dockerfileModalDisclosure = useDisclosure();
+  const [inputField, setInputField] = useState({url: ""})
```

Comment:
As you are using this only for storing the url, it's not necessary to create another dictionary inside it. You can use it directly to store the url